todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailFragment.java

```diff
@@ -90,24 +87,105 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         mDetailCompleteStatus = (CheckBox) root.findViewById(R.id.task_detail_complete);
 
         // Set up floating action button
-        FloatingActionButton fab =
-                (FloatingActionButton) getActivity().findViewById(R.id.fab_edit_task);
-
-        fab.setOnClickListener(__ -> mPresenter.editTask());
+        fab = (FloatingActionButton) getActivity().findViewById(R.id.fab_edit_task);
 
         return root;
     }
 
     @Override
-    public void setPresenter(@NonNull TaskDetailContract.Presenter presenter) {
-        mPresenter = checkNotNull(presenter);
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        mViewModel = ViewModelProviders.of(this, ToDoViewModelFactory.getInstance(getContext()))
+                .get(TaskDetailViewModel.class);
+        mDisposables = new CompositeDisposable();
+        bind();
+    }
+
+    private void bind() {
+        mDisposables.add(mViewModel.states().subscribe(this::render));
+        mViewModel.processIntents(intents());
+
+        RxView.clicks(fab).throttleFirst(200, TimeUnit.MILLISECONDS)
```

Comment:
Hmm I think I confused `throttleFirst` with `debounce`. I believe we should go for the second one, since we only want one item to be emitted after a specific time has passed and no other element has been emitted. What do you think?

Reply:
I'm not sure what the purpose is, are you using to avoid like consecutive clicks? Like a request would be sent twice or something like that?

Reply:
yes, avoiding multiple clicks